server/apps/main/views.py

```diff
@@ -49,44 +49,95 @@ def logout_user(request):
+            form = ShareExperienceForm(request.POST)
+            # check whether it's valid:
+            if edit:
+                # if edit=True, we render the share_experiences.html with form prepopulated.
+                return render(request, 'main/share_experiences.html', {'form': form})    
+            else:
+                # if edit=False, we proceed to submission.
+                if form.is_valid():
+                    
+                    # make a new uuid if doesn't already exist.
+                    uuid = form.cleaned_data.pop("uuid", False)
+                    
+                    if uuid: 
+                        # we will be here if we are editing a record already exists               
+
+                        # for OH we need to Delete before reupload.
+                        request.user.openhumansmember.delete_single_file(file_basename = f"{uuid}.json")
+                    
+                    else:
+                        uuid = make_uuid()
+                        
+                    upload(data = form.cleaned_data, uuid = uuid, ohmember = request.user.openhumansmember)                
+                    
+                    # for Public Experience we need to check if it's viewable and update accordingly.
+                    update_public_experience_db(data=form.cleaned_data, uuid=uuid, ohmember=request.user.openhumansmember)                
+                    
+                    # redirect to a new URL:
+                    return redirect('main:confirm_page')
+
+        # if a GET (or any other method) we'll create a blank form
+        else:
+            return render(request, 'main/share_experiences.html', {'form': ShareExperienceForm()})    
+    
+    else:    
+        return redirect('index')
 
-    tmp = OpenHumansMember.objects.all()
-    for i in tmp:
-        print(i)
+def update_public_experience_db(data, uuid, ohmember, moderation_status = 'not reviewed'):
+    """Updates the public experience database for the given uuid.
     
-    if request.method == 'POST':
+    If data is tagged as viewable, an experience will be updated or inserted.
+    If a data is tagged as not public, this function ensures that it is absent from the pe db.
 
-        print("HERE")
+    Args:
+        data (dict): an experience
+        uuid (str): unique identifier
+        ohmember : request.user.openhumansmember
+        moderation_status (str, optional): Defaults to 'in review'.
+    """
+    
+    if data['viewable']:
+        
+        pe = PublicExperience(experience_text=data['experience'],
+            difference_text=data['wish_different'],
+            title_text=data['title'],
+            open_humans_member=ohmember,
+            experience_id=uuid,
+            abuse=data['abuse'],
+            violence=data['violence'],
+            drug=data['drug'],
+            mentalhealth=data['mentalhealth'],
+            negbody=data['negbody'],
+            other=True if data['other'] != '' else False,
```

Comment:
This confuses me slightly, does it store the actual text of the 'other' label or just a binary True/False?